### DISCUSSION TITLE
Is there some advancement on "MultiText" to call all text in a single drawcall?

### DISCUSSION THREAD

**Discussion by KamyarTaher** [protectwise/troika#187] [General]
Hello, 

I recently came across your library troika-text which is really great!
But I still have one problem, we are forced to have several draw calls in order to generate different text on different positions.
But I always thought it would be possible to have merged geometries with a texture atlas for each part, and that these merged geometries are one instance.
Then I found "MultiText" on the issues which seem to do more or less that, but it seems to work only on the plane now. I don’t know if they were some advancements on the project? I think It would really help improve the performance of my project.

**Comment by lojjic** [comment]
I haven't made any more progress on it yet, unfortunately, though I've been doing other work around supporting multiple fonts that would be helpful with that.

I also don't yet have a great idea for how to allow non-coplanar text instances -- it seems that will essentially require a matrix4 per glyph which may not scale so well.

I'd gladly welcome assistance with it.

**Reply by KamyarTaher** [reply]
Thank you for your answer. For the moment I created a limited number of text objects that are assigned only to the positions of the objects that appear in front of the frustum. It's not perfect but it works.
I can't dwell on this problem for too long at the moment, but I'll gladly tackle it as soon as I have fewer obligations.

**Reply by lojjic** [reply]
> created a limited number of text objects that are assigned only to the positions of the objects that appear in front of the frustum

Text instances should be frustum culled automatically, by the way.
